.github/workflows/main.yml

```diff
@@ -0,0 +1,44 @@
+name: ci
+on: [push, pull_request]
+
+jobs:
+  build_and_deploy:
+    env:
+        BASE_VERSION: 5.0.0.2012
+    runs-on: windows-latest
+    steps:
+    - uses: actions/checkout@v1
+    - name: Download Nuget.exe
+      uses: NuGet/setup-nuget@v1.0.5
+      with:
+        nuget-version: 5.4.0
+    - name: setup-msbuild
+      uses: microsoft/setup-msbuild@v1
+    - name: Restore & Build
+      run: |
+        cd Xamarin.Forms.Platform.GTK
+        Nuget.exe restore Xamarin.Forms.Platform.GTK.csproj
+        msbuild Xamarin.Forms.Platform.GTK.csproj /property:Configuration=Release
+    - name: Package and upload
+      if: ${{ github.event_name == 'push' }}
+      run: |
+        $buildConfiguration = "Release"
+        $date = get-date -format "yyyyMMdd-HHmm"
+        $hash = & git rev-parse --short HEAD
+        # Generates a version number with format "{XF.Core base version}--date{date}.git-{commit-id}"
+        $version = "$env:BASE_VERSION--date$date.git-$hash" 
```

Comment:
you don't need to use powershell to generate this version number, you can use nugetPush.fsx with the flag `--output-version` so that it just prints the version (and you can store in a variable), instead of nugetPushing. An example is here: https://github.com/nblockchain/DotNetEssentials/blob/master/.github/workflows/ci.yml

Reply:
nugetPush.fsx generates problamatic version numbers (ex "5.0.0.2012.0--date20210503-1029.git-41b0e66") which are not valid semVer version. (the reason being the base version already has a .2012 trailing suffix)

Reply:
what should be the generated version then?

Reply:
5.0.0.2012--date20210503-1029.git-41b0e66

Reply:
ok we have to fix nugetPush.fsx to not add an extra `.0` when there are already 3 dots in the base version

Reply:
please propose PR for this ^ then I merge it, then you update this PR, then I merge it and then you can update geewallet's PR